common/data_refinery_common/models/models.py

```diff
@@ -1022,6 +1023,9 @@ def sync_from_s3(self, force=False, path=None):
         target_directory = os.path.dirname(self.absolute_file_path)
         os.makedirs(target_directory, exist_ok=True)
 
+        if not self.s3_bucket or not self.s3_key:
+            raise Exception('Tried to download a computed file with no s3_bucket or s3_key')
```

Comment:
Hmm, how about a KeyError or something instead of a raw Exception? You aren't really supposed to raise those directly.

Reply:
Good point, what if we raise [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError)? 

`KeyError` seems to be for dicts

Reply:
Good point, what if we raise ValueError? KeyError seems to be for dicts

Reply:
Sounds good to me!